src/py42/exceptions.py

```diff
@@ -5,36 +5,85 @@ class Py42Error(Exception):
     """A generic, Py42 custom base exception."""
 
 
-class Py42ArchiveFileNotFoundError(Py42Error):
+class Py42ResponseError(Py42Error):
```

Comment:
@timabrmsn I made some adjustments based on what you said -- had to make another base-y exception type here. This scenario is good for exceptions that are raised from responses that did not result in HTTPError, such as `UserDoesNotExistError` where the response is an empty list and we raise an error. Now, we can return the response for debugging purposes.

We still need `Py42Error` because there are some errors which there is not 1 single response responsible for causing the error, such the errors that get raised during `__setitem__` in the `Py42Response` or the error that gets raised when not finding a file (after paging over many responses) in the securitydata module.

Reply:
nice, I like it.